WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/Toolbar.kt

```
@@ -0,0 +1,67 @@
+package com.woocommerce.android.ui.compose.component
+
+import androidx.annotation.DrawableRes
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.woocommerce.android.R.drawable
+import com.woocommerce.android.R.string
+
+@Composable
+fun Toolbar(
+    onCloseButtonClick: (() -> Unit)?,
+    modifier: Modifier = Modifier,
+    onActionButtonClick: (() -> Unit)? = null,
+    title: String = "",
+    closeButtonIcon: ImageVector = Icons.Filled.ArrowBack,
+    @DrawableRes actionButtonIcon: Int = drawable.ic_help_24dp,
+    actionButtonText: String? = null
+) {
```

Thank you for stepping forward and extracting this to a reusable component @0nko 🙏🏼. A lot of duplication was removed thanks to this 🥇 

I think this is a good first step, and we can iterate over it as our needs for customizing grow. But I'd suggest a few changes to the current implementation:

- I think we should support multiple actions. For example `StoreProfilerCategoryScreen` has a "Skip" button but I think it would make sense to have the help button in that screen too. We can have an `actions` param such as this `actions: @Composable RowScope.() -> Unit = {}` to have more flexibility.
- I'm missing `contentDescription` fields for the actions in the toolbar. Some of the @Composable `Toolbars` that had been replaced in this PR had it and now are missing it. I'll mark those where I noticed content description is missing :) In most cases the content description can be addressed by a providing a default value. So it should not be too much effort to fix it. 
- I'd define `title` param as `title: @Composable () -> Unit,` rather than a string to keep it customizable
- If `onCloseButtonClick` is an optional param we should add null as default value, if not we should not make it nullable right?
-

---

I agree with most of the points @JorgeMucientes 👍.

> I'd define title param as title: @Composable () -> Unit, rather than a string to keep it customizable

This would be nice, but it would also be nice to have an overload that accepts just a `String`, this way we can keep the advantage of the simplicity of providing a single `String`, while allowing the customizability if someone needs it, WDYT?

And two additional remarks (np):
1. I think it would be better to control the visibility of the navigation icon using the icon itself rather than the callback, at least personally this would be clearer to me. If we follow this, then the Composable should throw an error if one of the arguments is null and the other is not.
2. Wouldn't the name `navigationIcon` be more aligned with the official components than `closeButtonIcon`? If someone is new to the component, and is trying the autocomplete to find the arguments, they would expect something like `navigationIcon` and `onNavigationClick` 🤔.

Anyway, these two last remarks from me are not blockers, just some thoughts, Jorge's points are the more important ones IMO.

---

Thanks for the reviews, @JorgeMucientes and @hichamboushaba! When I extracted the `Toolbar` I made it just as flexible to cover the existing use cases, but your suggestions are great, thank you!

It seems like there are 2 most common "types" of toolbars, so I can create overloads for those along with one all the options so we can benefit both from the simplicity and flexibility.

---

As per your suggestions, I've added a couple of `Toolbar` overloads in [140bf8b](https://github.com/woocommerce/woocommerce-android/pull/8054/commits/140bf8b83ad52d5e64c5d21ea5726e0f3c95b6d6). Basically, there's a more generic one with just the navigation icon and Composables for the `title` & `actions` and then 2 more with common default values both for a single icon button and text button. Also the action/navigation button now either must have both icon/title & handler `null` or an error is thrown.

Ready for another round @hichamboushaba!